src/language/server.ts

```diff
@@ -277,13 +282,61 @@ connection.onCompletionResolve(
 /**
  * Execute the entire playground script.
  */
-connection.onRequest('executeAll', async (params) => {
-  const connectionOptions = params.connectionOptions || {};
-  const runtime = new ElectronRuntime(
-    await CliServiceProvider.connect(params.connectionString, connectionOptions)
-  );
+connection.onRequest('executeAll', (params) => {
+  // Use Node worker threads to isolate each run of a playground
+  // to be able to cancel evaluation of infinite loops
+  const worker = new WorkerThreads(path.resolve(__dirname, 'worker.js'), {
```

Comment:
There is an issue with support for `.ts` files. Trying to run a `.ts` file in a worker thread results in error:
`The worker script extension must be “.js” or “.mjs”. Received “.ts”`

As a workaround, I require `.js` file from the out folder when creating a worker.

Reply:
Makes sense. I know you guys are going to use Webpack here, then maybe https://webpack.js.org/loaders/worker-loader/ or some similar 3rd-party plugin may help.

Reply:
Yes, we are going to webpackify the extension any time soon. Hope we will be able to configure it to get rid of this workaround. We also do compile before testing anyway, not because of this particular issue, and extension reads all files from the out folder when running. So probably this it not so much harm at this point. And do you use poor js workers only for tests in `mongosh`? The real `shell-api` uses the native workers, right?